library/src/G4LatticeLogical.cc

```diff
@@ -690,6 +691,148 @@ void G4LatticeLogical::AddValley(G4double phi, G4double theta, G4double psi) {
   fValleyAxis.push_back(fValleyInv.back()*G4ThreeVector(1.,0.,0.));
 }
 
+// Store drifting-electron valley using valley's direction
+
+void G4LatticeLogical::AddValley(const G4ThreeVector& valleyDirVec) {
+      
+  // Find the rotation matrix elements. 
```

Comment:
This description should be possible to make much shorter.

Reply:
I do not see any changes to the description.  The description should not simply repeat the actual calculations in the code.  It should be explaining *why* the code is written, or *how* the code does the job.  Pretend you're at a whiteboard explaining it to another student.  What words would you say to them while writing down the equations?

Reply:
I made a few changes such as shortening the explications, explaining that the matrix must be right-handed, the z axis must point upward for valleys and downward for anti-valleys, etc.). I'm not quite sure what else to change, I was trying to do something similar to ChargeCarrierTimeStep in branch G4CMP-374!